src/ui/hook/useDropDirection.ts

```diff
@@ -0,0 +1,23 @@
+import { useState, useEffect, useCallback, type RefObject } from 'react'
+
+export const useDropDirection = (ref: RefObject<HTMLElement>): boolean => {
+  const [dropup, setDropup] = useState(false)
+
+  const updateDropDirection = useCallback(() => {
+    if (ref.current) {
+      setDropup(ref.current.getBoundingClientRect().y > window.innerHeight / 2)
+    }
+  }, [ref])
+
+  useEffect(() => {
+    updateDropDirection()
+
+    window.addEventListener('resize', updateDropDirection)
```

Comment:
In your code, the state is updated immediately whenever the window is resized. It works, but resizing events can occur rapidly, potentially resulting in numerous state updates. That's why  I'd recommend debouncing the events first (a hook may help in such case).